ocean_provider/routes/README.md

```diff
@@ -0,0 +1,166 @@
+<!--
+Copyright 2021 Ocean Protocol Foundation
+SPDX-License-Identifier: Apache-2.0
+-->
+
+# Possible errors returned by Provider
+
+This document reflects couple of the possible errors returned by Provider.
+## Compute routes
+
+### initializeCompute
+
+In order to consume a data service the user is required to send
+one datatoken to the provider, as well as provider fees for the compute job.
+
+#### 1. 400 - Bad Request
+
+It occurs when the validation part fails.
+The following errors are displayed in JSON format:
+
+##### 1.1 For algorithm validation
+```python
+{
+    "additional_input": "invalid"
+}
+```
+**Reason** The `additional_inputs` key is not a list.
+
+```python
+{
+    "algorithm": "missing_meta_documentId"
+}
+```
+**Reason** Either algorithm metadata, either algorithm DID is missing.
+
+```python
+{
+    "algorithm": "file_unavailable"
+}
+```
+**Reason** One possibility is that the asset could not be retrieved from Aquarius's database.
+Otherwise, there are issues related to `services`, such as:
+- particular `service` is not found
+
+
+- `Key <key> not found in files.` - `datatokenAddress` or `nftAddress` or `files` is missing
+the decrypted files object
+
+
+- `Mismatch of datatoken.` - mismatch between service datatoken & decrypted files datatoken;
+
+
+- `Mismatch of dataNft.` - mismatch between asset data NFT address & the one from the decrypted files;
+
+
+- `Expected a files list` - `files` is not a list;
+
+
+- `Error decrypting service files` - other errors for decrypting the file.
+
+```python
+{
+    "algorithm": "not_algo"
+}
+```
+**Reason** The `type` from the algorithm's metadata is not specified as `algorithm`.
+
+
+```python
+{
+    "algorithm.serviceId": "missing"
+}
+```
+or
+
+```python
+{
+    "algorithm.serviceId": "not_found"
+}
+```
+
+**Reason** The `serviceId` key is missing from the algorithm's DDO.
+
+```python
+{
+    "algorithm.documentId": "did_not_found"
+}
+```
+
+**Reason** The asset `DID` could not be retrieved from the metadata store.
```

Comment:
please specify here that this is the algorithm's DID and the "algorithm asset" does not exist, to avoid unclarity regarding which asset is unretrievable